sdk/digitaltwins/Azure.DigitalTwins.Core/perf/Azure.DigitalTwins.Core.Perf/Scenarios/QueryDigitalTwins.cs

```diff
@@ -0,0 +1,96 @@
+﻿//Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.DigitalTwins.Core.Perf.Infrastructure;
+using Azure.Identity;
+using Azure.Test.Perf;
+using NUnit.Framework;
+
+namespace Azure.DigitalTwins.Core.Perf.Scenarios
+{
+    /// <summary>
+    /// The performance test scenario focused on running queries against digital twins instances.
+    /// </summary>
+    /// <seealso cref="PerfTest{SizeOptions}" />
+    public sealed class QueryDigitalTwins : PerfTest<SizeOptions>
+    {
+        private readonly DigitalTwinsClient _digitalTwinsClient;
+        private readonly string _testId;
+        private readonly long _size;
+        private readonly TimeSpan _delayPeriod = TimeSpan.FromMinutes(1);
+
+        public QueryDigitalTwins(SizeOptions options) : base(options)
+        {
+            _digitalTwinsClient = new DigitalTwinsClient(
+                new Uri(PerfTestEnvironment.Instance.DigitalTwinsUrl),
+                new ClientSecretCredential(
+                    PerfTestEnvironment.Instance.DigitalTwinsTenantId,
+                    PerfTestEnvironment.Instance.DigitalTwinsClientId,
+                    PerfTestEnvironment.Instance.DigitalTwinsClientSecret));
+
+            _size = options.Size;
+            _testId = Guid.NewGuid().ToString().Substring(0, 8);
+        }
+
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync();
+
+            // Global setup code that runs once at the beginning of test execution.
+            // Create the model globally so all tests can take advantage of it.
+            await AdtInstancePopulator.CreateRoomModelAsync(_digitalTwinsClient).ConfigureAwait(false);
+        }
+
+        public override async Task SetupAsync()
+        {
+            await base.SetupAsync();
+            await AdtInstancePopulator.CreateIndividualRoomTwins(_digitalTwinsClient, _testId, _size).ConfigureAwait(false);
+
+            // Since it takes some time for the newly created twins to be included in the query result, we have to wait some time.
+            await Task.Delay(_delayPeriod);
+        }
+
+        /// <summary>
+        /// Queries for all digital twins using <see cref="DigitalTwinsClient.Query{T}(string, CancellationToken)"/>.
+        /// </summary>
+        /// <param name="cancellationToken">The token used to signal cancellation request.</param>
+        public override void Run(CancellationToken cancellationToken)
+        {
+            Pageable<BasicDigitalTwin> result = _digitalTwinsClient
+                .Query<BasicDigitalTwin>($"SELECT * FROM DIGITALTWINS WHERE TestId = '{_testId}'", CancellationToken.None);
```

Comment:
we will only query for twins with the test Id that is associated with this object.